packages/shared_preferences/shared_preferences_android/lib/src/shared_preferences_async_android.dart

```diff
@@ -181,16 +229,53 @@ base class SharedPreferencesAsyncAndroid
     SharedPreferencesOptions options,
   ) async {
     final PreferencesFilters filter = parameters.filter;
-    final Map<String?, Object?> data = await _api.getAll(
+    final SharedPreferencesPigeonOptions pigeonOptions =
+        _convertOptionsToPigeonOptions(options);
+    final SharedPreferencesAsyncApi api = _getApiForBackend(pigeonOptions);
+    final Map<String?, Object?> data = await api.getAll(
       filter.allowList?.toList(),
-      _convertOptionsToPigeonOptions(options),
+      pigeonOptions,
     );
     return data.cast<String, Object>();
   }
 }
 
+/// Used to identify which Android library should be used on the backend of this call.
+enum SharedPreferencesAndroidBackendLibrary {
+  /// Represents the newer DataStore Preferences library.
+  DataStore,
+
+  /// Represents the older SharedPreferences library.
+  SharedPreferences,
+}
+
 /// Options for the Android specific SharedPreferences plugin.
 class SharedPreferencesAsyncAndroidOptions extends SharedPreferencesOptions {
   /// Constructor for SharedPreferencesAsyncAndroidOptions.
-  const SharedPreferencesAsyncAndroidOptions();
+  const SharedPreferencesAsyncAndroidOptions({
+    this.backend = SharedPreferencesAndroidBackendLibrary.DataStore,
+    this.originalSharedPreferencesOptions,
+  });
+
+  /// Which backend should be used with
+  final SharedPreferencesAndroidBackendLibrary backend;
+
+  /// These options define how the `SharedPreferences` library should behave.
```

Comment:
Maybe s/library/backend/ so you're not introducing a new term into the API docs here?